File: Demo/LGSideMenuControllerDemo/ChooseNavigationController.m
//
//  ChooseNavigationController.m
//  LGSideMenuControllerDemo
//
//  Created by Grigory Lutkov on 05.11.15.
//  Copyright (c) 2015 Grigory Lutkov. All rights reserved.
//

#import "ChooseNavigationController.h"

@interface ChooseNavigationController ()

@end

@implementation ChooseNavigationController

- (void)viewDidLoad
{
    [super viewDidLoad];

    self.navigationBar.translucent = YES;
    self.navigationBar.barTintColor = [UIColor colorWithRed:0.f green:0.5 blue:1.f alpha:1.f];
    self.navigationBar.titleTextAttributes = @{NSForegroundColorAttributeName: [UIColor whiteColor]};
    self.navigationBar.tintColor = [UIColor colorWithWhite:1.f alpha:0.5];
}

- (BOOL)shouldAutorotate
{
    return YES;
}

- (BOOL)prefersStatusBarHidden
{
    return (UIInterfaceOrientationIsLandscape([UIApplication sharedApplication].statusBarOrientation) && UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPhone);
}

- (UIStatusBarStyle)preferredStatusBarStyle
{
    return UIStatusBarStyleLightContent;
}

- (UIStatusBarAnimation)preferredStatusBarUpdateAnimation
{
    return UIStatusBarAnimationNone;
}

@end

File: Demo/Swift/LGSideMenuControllerDemo/ChooseNavigationController.swift
//
//  ChooseNavigationController.swift
//  LGSideMenuControllerDemo
//

class ChooseNavigationController: UINavigationController {
    
    init() {
        super.init(rootViewController: ChooseTableViewController())
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()

        navigationBar.barTintColor = UIColor(red: 0.0, green: 0.5, blue:1.0, alpha:1.0)
        navigationBar.titleTextAttributes = [.foregroundColor: UIColor.white]
    }
    
    override var shouldAutorotate : Bool {
        return true
    }
    
    override var prefersStatusBarHidden : Bool {
        return UIApplication.shared.statusBarOrientation.isLandscape && UIDevice.current.userInterfaceIdiom == .phone
    }
    
    override var preferredStatusBarStyle : UIStatusBarStyle {
        return .lightContent
    }
    
    override var preferredStatusBarUpdateAnimation : UIStatusBarAnimation {
        return .none
    }
    
    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
        super.traitCollectionDidChange(previousTraitCollection)
        self.setColors()
    }
    
    func setColors() {
        self.view.backgroundColor = (isLightTheme() ? .white : .black)
    }
}
